archive/frames/views.py

```diff
@@ -163,83 +168,174 @@ def zip(self, request):
+        # authenticated users each get their own cache space
+        if request.user.is_authenticated:
+            cache_key_elms.append(request.user.id)
+
+        cache_key = blake2b("".join(map(str, cache_key_elms)).encode("utf-8")).hexdigest()
+
+        response_dict = cache.get(cache_key)
+
+        if response_dict:
+            response_serializer = self.get_response_serializer(response_dict)
+            return Response(response_serializer.data)
+
+        frames = Frame.objects.all()
+
+        frames = frames.filter(observation_date__gte=start)
+
+        frames = frames.filter(observation_date__lt=end)
+
+        if not public:
+            logger.info("excluding public data")
+            frames = frames.exclude(public_date__lt=Now())
+
+        if site_id:
+            frames = frames.filter(site_id=site_id)
+
+        if telescope_id:
+            frames = frames.filter(telescope_id=telescope_id)
+
+        if primary_optical_element:
+            frames = frames.filter(primary_optical_element=primary_optical_element)
+
+        if instrument_id:
+            frames = frames.filter(instrument_id=instrument_id)
+
+        if configuration_type:
+            frames = frames.filter(configuration_type=configuration_type)
+
+        if proposal_id:
+            frames = frames.filter(proposal_id=proposal_id)
+
+        if not request.user.is_authenticated:
+            frames = frames.exclude(public_date__gte=Now())
+        elif not request.user.is_superuser:
+            user_proposals = request.user.profile.proposals or []
+
+            # Authenticated users can only aggregate over their non-public data
+            if user_proposals:
+                frames = frames.filter(
+                    public_date__gte=Now(),
+                    proposal_id__in=user_proposals
+                )
+            else:
+                logger.info(
+                    "No proposals found for user. "
+                    "Aggregating over only public data."
+                )
+                frames = frames.exclude(public_date__gte=Now())
+
+        try:
+            response_dict = aggregate_raw_sql(frames, timeout)
+        except OperationalError:
+            logger.info("filtered aggregation timed out")
+            return Response(
+                "Aggregation query timed out. Either try narrowing the search "
+                "space by adding more filters or increase the timeout.",
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+
+        if request.user.is_authenticated:
+            # 30 secs
+            # Don't expect a lot of hits, but might be good if they're going
+            # back and forth in short bursts of activity.
+            cache_timeout = 30
+        else:
+            # 15 mins
+            # unautheticated users share the same cache space making it much
+            # more likely they'll be hitting the same keys
+            cache_timeout = 15 * 60
+
```

Comment:
My thoughts are we might as well cache both of these a bit longer, since the sets of agg options should change infrequently with new data coming in. Maybe 1hr for authenticated, 24hr for non?

Reply:
I'm cautious about increasing the time for the authenticated users, because each user has their own cache. Stale data may end up saturating the cache making it much less effective. Authenticated users SQL aggregation also happens quicker because of the proposal ids filter, so I don't know if 1 hr is worth it. Let's try 5 mins first.

Non-authenticated I'm fine with 24 hours.